src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -193,13 +189,49 @@ object LoadVCF {
         s"""corrupt VCF: expected final header line of format `#CHROM\tPOS\tID...'
            |  found: @1""".stripMargin, headerLine)
 
-    val sampleIds: Array[String] =
-      if (dropSamples)
-        Array.empty
-      else
-        headerLine
-          .split("\t")
-          .drop(9)
+    val sampleIds: Array[String] = headerLine.split("\t").drop(9)
+
+    VCFHeaderInfo(sampleIds, infoSignature, vaSignature, gSignature, canonicalFlags, headerLines)
+  }
+
+  def apply(hc: HailContext,
+    reader: HtsjdkRecordReader,
+    file1: String,
+    files: Array[String],
+    nPartitions: Option[Int] = None,
+    dropSamples: Boolean = false): VariantSampleMatrix[Locus, Variant, Annotation] = {
+    val sc = hc.sc
+
+    val vcfHeaders: Array[VCFHeaderInfo] = files.map(parseHeader(hc, reader, _))
+    val header1 = vcfHeaders.head
+
+    files.zip(vcfHeaders).foreach { case (file, hd) =>
+      if (!(header1.sampleIds sameElements hd.sampleIds)) {
+        val firstDifference = header1.sampleIds.zip(hd.sampleIds).zipWithIndex.dropWhile { case ((s1, s2), i) => s1 == s2 }.head
+        fatal(
+          s"""invalid sample ids: expected sample ids to be identical for all inputs.
+           |    Found different sample ids at position ${ firstDifference._2 + 1 }
+           |    ${ files(0) }: ${ firstDifference._1._1 }
+           |    $file: ${ firstDifference._1._2 }""".stripMargin)
+      }
+
+      if (header1.genotypeSignature != hd.genotypeSignature)
+        fatal(s"""invalid genotype signature: expected signatures to be identical for all inputs.
+               |   ${ files(0) }: ${ header1.genotypeSignature.toPrettyString() }
```

Comment:
also probably attributes = true, because this is going to fail if the field descriptions are different